fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/query/LookbackQuery.java

```diff
@@ -266,7 +266,20 @@ public LookbackQuery withFilter(Filter filter) {
 
     @Override
     public LookbackQuery withIntervals(Collection<Interval> intervals) {
-        return withDataSource(new QueryDataSource(getInnerQueryUnchecked().withIntervals(intervals)));
+        return new LookbackQuery(
+                new QueryDataSource(getInnerQueryUnchecked().withIntervals(intervals)),
+                granularity,
+                filter,
+                aggregations,
+                postAggregations,
+                intervals,
+                context,
+                true,
+                lookbackOffsets,
+                lookbackPrefixes,
+                having,
+                limitSpec
+        );
```

Comment:
I believe a `wither` is supposed to mutate a single attribute of an object.

Could we have a new `wither` for `doFork`? You can chain the two withers together for your purpose

Reply:
Added a new `wither` and modified the change log